src/components/script/ScriptEditor.js

```diff
@@ -138,8 +138,10 @@ class ActionMini extends Component {
 
   onPasteEvent = before => e => {
     const { id, onPaste } = this.props;
-    const { clipboardEvent } = this.state;
-    onPaste(id, clipboardEvent, before);
+    const clipboardEvent = this.readClipboard();
+    if (clipboardEvent) {
```

Comment:
The app crashes when pasting a Custom Event Script via the new `Paste Event` or `Paste Event Before/After`. I think it was crashing before, but changing this line to 

```if (clipboardEvent && clipboardEvent.__type === "event")``` 

should fix it.

PS: I don't think I can add changes in this PR. If you prefer I can do a separate one with this change.

Reply:
Oooh, good spot! So actually this wasn't an issue with custom events but with copying any entire script and pasting it where you could paste a single event, this did actually work until my refactor in the last few days. I think the change you suggested would stop the errors but prevent you from being able to paste a full script.

Instead I made this change 5d157ba07cf207c9969ddde6081f954329d8b87a line 177 is the important one, when a full script is in the clipboard you need to read the array of events out the of the `script` field before pasting.

Thanks for your help!